### PR TITLE
feat(cli): add async assignments and status of group-resource assignment

### PR DESCRIPTION
**Pull request by cuadradek**
- New objects were added to perun-cli: AssignedGroup, AssignedResource,
EnrichedGroup, EnrichedResource.
- Async option was added to tool assignGroupToResource to assign groups
asynchronously.
- Tools listOfGroupResources and listOfGroupsAssignedToResource now
also print status of group-resource assignments, source group id if it
is automatic assignment and whether the group's subgroups are
automatically assign as well.

### CODE REVIEW

File: perun-cli/Perun/beans/AssignedResource.pm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	return Perun::Common::fromHash(@_);
+}
+
+sub TO_JSON
+{
+	my $self = shift;
+
+	return { enrichedResource => $self->{_enrichedResource}, status => $self->{_status},
+		sourceGroupId => $self->{_sourceGroupId}, failureCause => $self->{_failureCause},
+		facility => $self->{_facility}, resourceTags => $self->{_resourceTags} };
```
Comment (Johaney-s) [inline perun-cli/Perun/beans/AssignedResource.pm:25]: Same here.

File: perun-cli/Perun/beans/EnrichedResource.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+sub getCommonArrayRepresentation {
+	my $self = shift;
+	return ($self->{_resource}->{id}, $self->{_resource}->{voId}, $self->{_resource}->{name},
+		$self->{_resource}->{facilityId}, $self->{_resource}->{description});
+}
+
+sub getCommonArrayRepresentationHeading {
+	return ('ID', 'Name', 'VO ID', 'Facility ID', 'Description');
```
Comment (Johaney-s) [inline perun-cli/Perun/beans/EnrichedResource.pm:42]:
```suggestion
	return ('ID', 'VO ID', 'Name', 'Facility ID', 'Description');
```

File: perun-cli/assignGroupToResource [hunk 3]
```diff
@@ -16,20 +16,21 @@ sub help {
...
 	--resource    | -r resource id
+	--async       | -a assigns group asynchronously
 	--batch       | -b batch
 	--help        | -h prints this help
 
 	};
 }
 
-my ($groupId, $groupName, $voId, $voShortName, $resourceId, $batch);
+my ($groupId, $groupName, $voId, $voShortName, $resourceId, $async, $batch);
```
Comment (Johaney-s) [inline perun-cli/assignGroupToResource:26]: `assignInactive` and `autoAssignSubgroups` are missing

**Review by Johaney-s — CHANGES_REQUESTED** [review]

File: perun-cli/Perun/beans/AssignedGroup.pm [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+{
+	return Perun::Common::fromHash(@_);
+}
+
+sub TO_JSON
+{
+	my $self = shift;
+
+	return { enrichedGroup => $self->{_enrichedGroup}, status => $self->{_status},
+		sourceGroupId => $self->{_sourceGroupId}, failureCause => $self->{_failureCause} };
```
[the remark this answers is not in the document]
Comment (cuadradek) [reply perun-cli/Perun/beans/AssignedGroup.pm]: Done.

File: perun-cli/Perun/beans/AssignedResource.pm (hunk 1 above)
Comment (cuadradek) [reply perun-cli/Perun/beans/AssignedResource.pm]: Done.

File: perun-cli/Perun/beans/EnrichedResource.pm (hunk 2 above)
Comment (cuadradek) [reply perun-cli/Perun/beans/EnrichedResource.pm:42]: Done, I fixed it in Resource.pm as well.

File: perun-cli/assignGroupToResource (hunk 3 above)
Comment (cuadradek) [reply perun-cli/assignGroupToResource]: Done.

**Review by Johaney-s — APPROVED** [review]

**Review by metodej — APPROVED** [review]
The code looks good.

**Pull request merged**